### ISSUE TITLE
repeat() fails when it rendered conditionally

### ISSUE THREAD

**Issue by surma** [PolymerLabs/lit-html#91]
I’m not sure this is a bug, but consider this test case I wrote:

```js
    test('can be rendered conditionally ', () => {
      let flag = false;
      const items = [1, 2, 3];
      const t = () => html`${flag ? repeat(items, (i: number) => html`<li>item: ${i}</li>`) : ''}`;
      render(t(), container);
      assert.equal(container.innerHTML, ``);
      flag = true;
      render(t(), container);
      assert.equal(container.innerHTML, `<li>item: 1</li><li>item: 2</li><li>item: 3</li>`);
    });
```

I know this can be fixed by moving the conditional to the iterable like `flag ? items : []` or something, but I am wondering if this should work too.

**Comment by justinfagnani** [comment]
Definitely a bug, but not related to repeat() specifically, but in the way directive return values are handled. It's fixed in #118 which ignores directive returns.

**Comment by justinfagnani** [comment]
This should be be fixed now!
